src/main/java/org/prebid/server/bidder/tappx/TappxBidder.java

```diff
@@ -0,0 +1,163 @@
+package org.prebid.server.bidder.tappx;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.Json;
+import org.apache.commons.lang3.StringUtils;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.tappx.ExtImpTappx;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class TappxBidder implements Bidder<BidRequest> {
+
+    private static final String VERSION = "1.0";
+
+    private static final TypeReference<ExtPrebid<?, ExtImpTappx>> TAPX_EXT_TYPE_REFERENCE =
+            new TypeReference<ExtPrebid<?, ExtImpTappx>>() {
+            };
+    private static final String DEFAULT_BID_CURRENCY = "USD";
+
+    private final String endpointUrl;
+
+    public TappxBidder(String endpointUrl) {
+        final String validatedEndpoint = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
```

Comment:
Should we validate url at this step? Since we know `endpointUrl` contains protocol only?

Reply:
I wanted to keep it there, bc someone can forgot about validation, after endpoint is changed.

If you insist, I can remove validation, but we (bidder) need to know about configuration.

Reply:
Yes, please.

Reply:
Done